src/server.rs

```diff
@@ -166,10 +271,12 @@ impl<L, M> Server<L, M>
                  -> Result<JoinHandle<Result<()>>> {
         thread::Builder::new()
             .name(format!("raft::Server({})", id))
-            .spawn(move || Server::run(id, addr, peers, store, state_machine))
+            .spawn(move || {
+                let mut server = try!(Server::create(id, addr, peers, store, state_machine, 1500, 3000, 1000, 129));
+                server.run()
+            })
```

Comment:
No problem. :) If you'd like to do structural changes I'm quite fine reviewing/accepting them. The code needs a lot of love and I haven't really been giving it. :(